src/Firebase/Auth/ApiClient.php

```diff
@@ -297,6 +298,25 @@ public function unlinkProvider(string $uid, array $providers): ResponseInterface
         ]);
     }
 
+    /**
+     * @throws AuthException
+     * @throws FirebaseException
+     */
+    public function signInWithOauthCredentials(Provider $provider, string $token, bool $isAccessToken = true)
```

Comment:
In the context of the SDK, even if it's just an internal method call, I think this should be called "linkProvider*` similar to the methods in the Auth class, and probably also should be duplicated for clarity's sake (I personally don't care about code duplication here and there :))